addons/mrp/models/mrp_workcenter.py

```diff
@@ -292,6 +292,13 @@ def _get_capacity(self, product):
         product_capacity = self.capacity_ids.filtered(lambda capacity: capacity.product_id == product)
         return product_capacity.capacity if product_capacity else self.default_capacity
 
+    def _get_product_expected_duration(self, product_id):
+        """In case there are specific capacities defined in the workcenter
+        that matches the product we are producing. Add the extra-time.
+        """
+        product = self.capacity_ids.filtered(lambda p: p.product_id == product_id)
+        return product.time_start + product.time_stop if product else 0.0
```

Comment:
Two little comments:
- I think the explanation is one sentence, not two ?
- `product` isn't a product, so it's a little dubious to name it like this :p
```suggestion
        that matches the product we are producing, adds the extra-time.
        """
        capacity = self.capacity_ids.filtered(lambda p: p.product_id == product_id)
        return capacity.time_start + capacity.time_stop if capacity else 0.0
```

Also, as we always use the workcenter's start and stop time, shouldn't be better to include them in this method to clean up the duration's calculation ?

Something like that:
```python
def _get_expected_duration(self, product_id):
    duration = self.time_start + self.time_stop
    capacity = self.capacity_ids.filtered(lambda p: p.product_id == product_id)
    duration += capacity.time_start + capacity.time_stop if capacity else 0.0
    return duration
```
... and then, adpat the code when we call this method (for instance :)
```python
return self.workcenter_id._get_expected_duration(self.product_id) + duration_expected_working * ratio * 100.0 / self.workcenter_id.time_efficiency
```

Reply:
- Actually, I think name them capacity would be more confusing but well the field are capacities after all -> Fixed in 48b9bed
- The _get_expected_duration suggestion was ingenious, thanks but we won't even need the variable duration (added in 48b9bed)